models/issue_tracked_time.go

```diff
@@ -74,34 +132,69 @@ func (opts *FindTrackedTimesOptions) ToSession(e Engine) *xorm.Session {
 	return x.Where(opts.ToCond())
 }
 
-// GetTrackedTimes returns all tracked times that fit to the given options.
-func GetTrackedTimes(options FindTrackedTimesOptions) (trackedTimes []*TrackedTime, err error) {
-	err = options.ToSession(x).Find(&trackedTimes)
+func getTrackedTimes(e Engine, options FindTrackedTimesOptions) (trackedTimes TrackedTimeList, err error) {
+	err = options.ToSession(e).Find(&trackedTimes)
 	return
 }
 
+// GetTrackedTimes returns all tracked times that fit to the given options.
+func GetTrackedTimes(opts FindTrackedTimesOptions) (TrackedTimeList, error) {
+	return getTrackedTimes(x, opts)
+}
+
+func getTrackedSeconds(e Engine, opts FindTrackedTimesOptions) (trackedSeconds int64, err error) {
+	return opts.ToSession(e).SumInt(&TrackedTime{}, "time")
+}
+
+// GetTrackedSeconds return sum of seconds
+func GetTrackedSeconds(opts FindTrackedTimesOptions) (int64, error) {
+	return getTrackedSeconds(x, opts)
+}
+
 // AddTime will add the given time (in seconds) to the issue
-func AddTime(user *User, issue *Issue, time int64) (*TrackedTime, error) {
-	tt := &TrackedTime{
-		IssueID: issue.ID,
-		UserID:  user.ID,
-		Time:    time,
-	}
-	if _, err := x.Insert(tt); err != nil {
+func AddTime(user *User, issue *Issue, amount int64, created time.Time) (*TrackedTime, error) {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
 		return nil, err
 	}
-	if err := issue.loadRepo(x); err != nil {
+
+	t, err := addTime(x, user, issue, amount, created)
```

Comment:
`addTime(sess`

Reply:
o missed that in the junglebof diffs ...

i'll 👀  again too ...

```suggestion
	t, err := addTime(sess, user, issue, amount, created)
```